### PR TITLE
Issue #119:  Excluding hsqldb-*.jar that comes from the prebuilt artifacts

### PR DESCRIPTION
**Pull request by jonathanmtran**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change

Excluding hsqldb-*.jar that comes from the prebuilt artifacts. This addresses #119

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
:thinking: uPortal 5.1.0+ has this applied by default https://github.com/Jasig/uPortal/commit/5f96aff623489d79c1e9494b36bb5828fdc2539c
This could be a good stopgap, thoughts @drewwills?

File: overlays/uPortal/build.gradle [hunk 1]
```diff
@@ -21,6 +21,7 @@ node {
 
 war {
     archiveName 'uPortal.war'
+    exclude 'WEB-INF/lib/hsqldb-*.jar'
```
Comment (drewwills) [inline overlays/uPortal/build.gradle]: This config matches how this issue is being handled in other overlays like Announcements and Calendar.

Fair enough -- having it here will (AFAIK) allow you to use uPortal 5.0.0 - 5.0.5 with up-to-date uPortal-start.  Perhaps it should be included.

But (IMHO) the solution we really want -- and the solution that was already applied to 5.1.x -- is to remove database driver jars from the builds of the war files that we overlay entirely.  They should all be marked `<scope>provided</scope>` (in Maven) or `providedRuntime` (in Gradle).  Database driver jars should always come from uPortal-start.

With that fix in place, we should -- at some point -- remove these lines.  For whatever period we continue to have these lines, we should add a comment WRT (1) what this line is for, (2) that there's a preferred solution percolating through the system, and (3) that these lines should be removed when the preferred solution is reliably in place.

File: overlays/jasig-widget-portlets/build.gradle [hunk 2]
```diff
@@ -8,4 +8,5 @@ dependencies {
 
 war {
     archiveName 'jasig-widget-portlets.war'
+    exclude 'WEB-INF/lib/hsqldb-*.jar'
```
Comment (drewwills) [inline overlays/jasig-widget-portlets/build.gradle]: So it looks like jasig-widget-portlets uses a database connection -- I'm not sure that was remembered when this file was set up.  If you're going to have this line, you also need `compile configurations.jdbc` within the `dependencies {}` section (of this build.gradle).

(Also see note on overlays/uPortal/build.gradle)

**Comment by drewwills** [comment]
@jonathanmtran -- This is a good catch.  It's immensely helpful to have more people in the community who can effectively:

  - Investigate the nature of issues when they are encountered (and we all encounter them)
  - Bring the kernel of an elegant solution forward in a pull request

Nicely done.  :+1:

**Comment by jonathanmtran** [comment]
Thanks for the feedback @drewwills  and @ChristianMurphy !

The branch has been updated to include the suggested changes. Feel free to edit the comment as necessary.

**Comment by drewwills** [comment]
I tweaked the comment just a bit (mostly so I could use the same wording in each project).

I applied the comment to every instance of `exclude 'WEB-INF/lib/hsqldb-*.jar'`.

**Pull request merged**